src/sas/sascalc/pr/p_invertor.py

```diff
@@ -556,59 +542,58 @@ def _get_matrix(self, nfunc, nr):
         b_obj = np.zeros(self.npoints + nr)
 
         sqrt_alpha = np.sqrt(self.alpha)
-        pi = np.pi
-        offset = (1, 0)[self.est_bck == 1]
+        offset = 0 if self.est_bck == 1 else 1
 
         if self.check_for_zero(self.err):
             raise RuntimeError("Pinvertor.get_matrix: Some I(Q) points have no error.")
 
-        #Compute A
-        #Whether or not to use ortho_transformed_smeared.
-        smeared = False
-        if self.slit_width > 0 or self.slit_height > 0:
-            smeared = True
+        # Whether or not to use ortho_transformed_smeared.
+        smeared = (self.slit_width > 0 or self.slit_height > 0)
+        smear_npts = 21
 
-        npts = 21
-        #Get accept_q vector across all q.
+        # Get valid points as x, y and err.
         q_accept_x = self.accept_q(self.x)
-
-        #The x and a that will be used for the first part of 'a' calculation, given to ortho_transformed
         x_use = self.x[q_accept_x]
-        a_use = a_obj[0:self.npoints, :]
+        y_use = self.y[q_accept_x]
+        err_use = self.err[q_accept_x]
 
+        # Create views into the a_obj/b_obj arrays containing the first
+        # n rows.  Since these are views into the original array updating
+        # the values will update the original.
+        a_view = a_obj[0:self.npoints, :]
+        b_view = b_obj[0:self.npoints]
+
+        # Compute A
         for j in range(nfunc):
             if self.est_bck == 1 and j == 0:
-                a_use[q_accept_x, j] = 1.0/self.err[q_accept_x]
+                res = 1.0
             elif smeared:
-                a_use[q_accept_x, j] = calc.ortho_transformed_smeared(x_use, self.d_max, j+offset,
-                                                                      self.slit_height, self.slit_width, npts)/self.err[q_accept_x]
+                res = calc.ortho_transformed_smeared(
+                    x_use, self.d_max, j+offset,
+                    self.slit_height, self.slit_width, smear_npts)
             else:
-                a_use[q_accept_x, j] = calc.ortho_transformed(x_use, self.d_max, j+offset)/self.err[q_accept_x]
-
-        a_obj[0:self.npoints, :] = a_use
+                res = calc.ortho_transformed(x_use, self.d_max, j+offset)
+            a_view[q_accept_x, j] = res/err_use
```

Comment:
I think it makes sense that both smeared and un-smeared is divided by error (that wasn't the cas before). Just make sure this something intended.

Reply:
This was the case before: all three branches used `.../self.err[q_accept_x]`.  I was just trying to make it more obvious.